plugins/gradle/src/org/jetbrains/plugins/gradle/service/task/GradleTaskManager.java

```diff
@@ -107,10 +115,16 @@ public void executeTasks(final @NotNull ExternalSystemTaskId id,
         CancellationToken cancellationToken = cancellationTokenSource.token();
         myHelper.ensureInstalledWrapper(id, rootProjectPath, effectiveSettings, listener, cancellationToken);
       }
-      myHelper.execute(projectPath, effectiveSettings, id, listener, cancellationTokenSource, connection -> {
-        executeTasks(id, tasks, projectPath, effectiveSettings, jvmParametersSetup, listener, connection, cancellationTokenSource);
-        return null;
-      });
+
+      GradleProgressIndicatorWrapper progressIndicatorWrapper = new GradleProgressIndicatorWrapper(id, effectiveSettings, listener);
```

Comment:
I am not sure if we need the wrapper and a new progress indicator here. The proper build event that you produce in GradleProgressListener should be enough.
Was it a workaround for some sort of issue?

Reply:
Without that wrapper progress indicator is not shown when running tests or tasks. Only a progress above "Build" window is shown:
<img width="1969" alt="Screenshot 2022-11-25 at 17 52 24" src="https://user-images.githubusercontent.com/3939893/204028957-17deb5c2-4230-4089-8a70-fb1c07493d4a.png">

Reply:
no, I guess it is an expected behavior.